azurerm/internal/services/monitor/monitor_activity_log_alert_resource.go

```diff
@@ -312,98 +355,143 @@ func resourceMonitorActivityLogAlertDelete(d *schema.ResourceData, meta interfac
 		})
 	}
 	if resourceGroup := v["resource_group"].(string); resourceGroup != "" {
-		conditions = append(conditions, insights.ActivityLogAlertLeafCondition{
+		conditions = append(conditions, insights.AlertRuleAnyOfOrLeafCondition{
 			Field:  utils.String("resourceGroup"),
 			Equals: utils.String(resourceGroup),
 		})
 	}
 	if id := v["resource_id"].(string); id != "" {
-		conditions = append(conditions, insights.ActivityLogAlertLeafCondition{
+		conditions = append(conditions, insights.AlertRuleAnyOfOrLeafCondition{
 			Field:  utils.String("resourceId"),
 			Equals: utils.String(id),
 		})
 	}
 	if status := v["status"].(string); status != "" {
-		conditions = append(conditions, insights.ActivityLogAlertLeafCondition{
+		conditions = append(conditions, insights.AlertRuleAnyOfOrLeafCondition{
 			Field:  utils.String("status"),
 			Equals: utils.String(status),
 		})
 	}
 	if subStatus := v["sub_status"].(string); subStatus != "" {
-		conditions = append(conditions, insights.ActivityLogAlertLeafCondition{
+		conditions = append(conditions, insights.AlertRuleAnyOfOrLeafCondition{
 			Field:  utils.String("subStatus"),
 			Equals: utils.String(subStatus),
 		})
 	}
 	if recommendationType := v["recommendation_type"].(string); recommendationType != "" {
-		conditions = append(conditions, insights.ActivityLogAlertLeafCondition{
+		conditions = append(conditions, insights.AlertRuleAnyOfOrLeafCondition{
 			Field:  utils.String("properties.recommendationType"),
 			Equals: utils.String(recommendationType),
 		})
 	}
 
 	if recommendationCategory := v["recommendation_category"].(string); recommendationCategory != "" {
-		conditions = append(conditions, insights.ActivityLogAlertLeafCondition{
+		conditions = append(conditions, insights.AlertRuleAnyOfOrLeafCondition{
 			Field:  utils.String("properties.recommendationCategory"),
 			Equals: utils.String(recommendationCategory),
 		})
 	}
 
 	if recommendationImpact := v["recommendation_impact"].(string); recommendationImpact != "" {
-		conditions = append(conditions, insights.ActivityLogAlertLeafCondition{
+		conditions = append(conditions, insights.AlertRuleAnyOfOrLeafCondition{
 			Field:  utils.String("properties.recommendationImpact"),
 			Equals: utils.String(recommendationImpact),
 		})
 	}
 
-	return &insights.ActivityLogAlertAllOfCondition{
+	if serviceHealth := v["servicehealth"].([]interface{}); len(serviceHealth) > 0 {
+		for _, serviceItem := range serviceHealth {
+			vs := serviceItem.(map[string]interface{})
+			if regions, ok := vs["regions"]; ok {
+				if rv, ok := regions.(*schema.Set); ok {
+					if len(rv.List()) > 0 {
+						conditions = append(conditions, insights.AlertRuleAnyOfOrLeafCondition{
+							Field:       utils.String("properties.impactedServices[*].ImpactedRegions[*].RegionName"),
+							ContainsAny: utils.ExpandStringSlice(rv.List()),
+						})
+					}
+				}
+			}
+
+			if events, ok := vs["events"]; ok {
+				if ev, ok := events.(*schema.Set); ok {
```

Comment:
Similar here, we could omit this because this has been ensured by the schema

Reply:
thx